src/com/google/enterprise/adaptor/database/TupleReader.java

```diff
@@ -205,45 +289,53 @@ protected void emitRow(ResultSet resultSet, ContentHandler handler)
         case Types.CHAR:
         case Types.VARCHAR:
           String string = resultSet.getString(col);
-          handler.startElement("", columnName, columnName, atts);
-          outWriter.write(string);
-          break;
-        case Types.CLOB:
-        case Types.LONGVARCHAR:
-          Reader reader = resultSet.getCharacterStream(col);
-          handler.startElement("", columnName, columnName, atts);
-          copyValidXMLCharacters(reader, outWriter);
-          reader.close();
+          if (string != null) {
+            try (Writer writer =
+                new BufferedWriter(new ContentHandlerWriter(handler))) {
+              handler.startElement("", columnName, columnName, atts);
+              writer.write(string);
+            }
+          }
           break;
         case Types.NCHAR:
```

Comment:
Maybe we should be calling getNString on these first two, but we're using getString elsewhere, and I think we should do the same here (especially since we changed CHAR/VARCHAR to avoid using a stream).